lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -286,5 +289,101 @@ def with_truncated_backtrace(child)
+      # code to detect exceptions that are instances of either, without first
+      # checking to see if rspec-expectations is loaded.
+      module InterfaceTag
+        # Appends the provided exception to the list.
+        # @param exception [Exception] Exception to append to the list.
+        # @private
+        def add(exception)
+          # `PendingExampleFixedError` can be assigned to an example that initially has no
+          # failures, but when the `aggregate_failures` around hook completes, it notifies of
+          # a failure. If we do not ignore `PendingExampleFixedError` it would be surfaced to
+          # the user as part of a multiple exception error, which is undesirable. While it's
+          # pretty weird we handle this here, it's the best solution I've been able to come
+          # up with, and `PendingExampleFixedError` always represents the _lack_ of any exception
+          # so clearly when we are transitioning to a `MultipleExceptionError`, it makes sense to
+          # ignore it.
+          return if Pending::PendingExampleFixedError === exception
+
+          all_exceptions << exception
+
+          if exception.class.name =~ /RSpec/
+            failures << exception
+          else
+            other_errors << exception
+          end
+        end
+
+        # Provides a way to force `ex` to be something that satisfies the multiple
+        # exception error interface. If it already satisfies it, it will be returned;
+        # otherwise it will wrap it in a `MultipleExceptionError`.
+        # @private
+        def self.for(ex)
+          return ex if self === ex
+          MultipleExceptionError.new(ex)
+        end
+      end
+
+      include InterfaceTag
+
+      # @return [Array<Exception>] The list of failures.
+      attr_reader :failures
+
+      # @return [Array<Exception>] The list of other errors.
+      attr_reader :other_errors
+
+      # @return [Array<Exception>] The list of failures and other exceptions, combined.
+      attr_reader :all_exceptions
+
+      # @return [Hash] Metadata used by RSpec for formatting purposes.
+      attr_reader :aggregation_metadata
+
+      # @return [nil] Provided only for interface compatibility with
+      #   `RSpec::Expectations::MultipleExpectationsNotMetError`.
+      attr_reader :aggregation_block_label
+
+      # @param exceptions [Array<Exception>] The initial list of exceptions.
+      def initialize(*exceptions)
+        super()
+
+        @failures                = []
+        @other_errors            = []
+        @all_exceptions          = []
+        @aggregation_metadata    = { :hide_backtrace => true }
+        @aggregation_block_label = nil
+
+        exceptions.each { |e| add e }
+      end
+
+      # @return [String] Combines all the exception messages into a single string.
+      # @note RSpec does not actually use this -- instead it formats each exception
+      #   individually.
```

Comment:
Seems wrong for an exception to not provide a `message`.  This exception may be caught by users and it would be crappy for us to return `nil` from `message` when they try to print the exception.  I didn't want to put any effort into a good formatting for `message` here but figured a simple `join("\n\n")` was a good way to provide a low-effort message.